src/cmap/auth/mongo_credentials.ts

```diff
@@ -149,7 +149,7 @@ export class MongoCredentials {
         return;
       }
       // TODO(NODE-3483): Replace this with a MongoAuthValidationError
-      throw new MongoDriverError(`Password not allowed for mechanism MONGODB-X509`);
+      throw new MongoRuntimeError(`Password not allowed for mechanism MONGODB-X509`);
```

Comment:
API error, someone would need to request MONGODB-X509 auth and provide a password. x509 relies on the TLS cert you provide so we prevent confusion by disallowing a password.